recipes/newrelic/infrastructure/cloud/aws-linux.yml

```diff
@@ -39,7 +132,112 @@ install:
+                              "sts:ExternalId": "{{.NEW_RELIC_ACCOUNT_ID}}"
+                          }
+                      }
+                  }
+              ]
+          }
+          EOT
+          )
+
+          createRole=$(aws iam create-role --role-name $roleName --assume-role-policy-document file://<(echo $policy) 2>&1)
+
+          getRole=$(aws iam get-role --role-name $roleName 2>&1)
+
+          if [ "$(echo "$getRole" | grep "^An error occurred (NoSuchEntity)*" | wc -l)" -eq 1 ] ; then
+              echo "$createRole" >&2
+              exit 1
+          fi
+
+          aws iam attach-role-policy --role-name $roleName --policy-arn arn:aws:iam::aws:policy/ReadOnlyAccess
+
+          policy=$(cat <<EOT
+          {
+              "Version": "2012-10-17",
+              "Statement": [
+                  {
+                  "Effect": "Allow",
+                  "Action": [
+                      "budgets:ViewBudget"
+                  ],
+                  "Resource": "*"
+                  }
+              ]
+          }
+          EOT
+          )
+
+          aws iam put-role-policy --role-name $roleName --policy-name NewRelicBudget --policy-document file://<(echo $policy)
+
+          callerIdentity=$(aws sts get-caller-identity 2>&1)
+
+          accountName=$(echo "$callerIdentity" | sed -ne 's/.*Account": "\(.*\)",/\1/p')
+
+          # Download and fix CloudFormation template
+          curl -sLo /tmp/MetricStreams_CloudFormation.yml {{.NEW_RELIC_DOWNLOAD_URL}}cloud_integrations/aws/cloudformation/MetricStreams_CloudFormation.yml
+          sed -ie "s/AWS::AccountId,/\'AWS::AccountId\',/g" /tmp/MetricStreams_CloudFormation.yml
+
+          printf "\nDeploying CloudFormation template, this may take a few minutes...\n"
+          deploy=$(aws cloudformation deploy --template-file /tmp/MetricStreams_CloudFormation.yml --stack-name NewRelic-Metric-Stream --parameter-overrides NewRelicLicenseKey={{.NEW_RELIC_LICENSE_KEY}} --capabilities CAPABILITY_NAMED_IAM 2>&1)
+
+          if [ "$(echo "$deploy" | grep "^An error occurred (AlreadyExistsException)*" | wc -l)" -eq 1 ] ; then
+              echo "CloudFormation stack already exists"
+              rm /tmp/MetricStreams_CloudFormation.yml
+              exit 0
+          fi
+
+          if [ "$(echo "$deploy" | grep "^An error occurred (InvalidChangeSetStatus)*" | wc -l)" -eq 1 ] ; then
+              echo "CloudFormation stack is already being deployed"
+              rm /tmp/MetricStreams_CloudFormation.yml
+              exit 0
+          fi
+
+          if [ "$(echo "$deploy" | grep "*CREATE_IN_PROGRESS*" | wc -l)" -eq 1 ] ; then
+              echo "CloudFormation stack is already being deployed"
+              rm /tmp/MetricStreams_CloudFormation.yml
+              exit 0
+          fi
+
+          echo "CloudFormation deployment complete"
+
+          rm /tmp/MetricStreams_CloudFormation.yml
+
+          getRole=$(aws iam get-role --role-name $roleName 2>&1)
```

Comment:
@noahmmcgivern we already did a getRole above, should we re-use the previous getRole variable instead of looking up the role again?